site/docs/row-level-deletes.md

```diff
@@ -0,0 +1,196 @@
+
+```
+file D: (2, 'c1', 'data2'), (4, 'c2', 'data2')
+file C: (5, 'c3', 'data3'), (6, 'c3', 'data2')
+```
+
+There is no effect on read side in the copy-on-write approach.
+
+## Merge-on-read
+
+### Row-level delete file spec
+
+Iceberg supports 2 different types of row-level delete files: **position deletes** and **equality deletes**.
+If you are unfamiliar with these concepts, please read the [spec](../spec/#row-level-deletes) page for more information before proceeding.
+
+Also note that because row-level delete files are valid Iceberg data files, all rows in each delete file must belong to the same partition.
+If a delete file belongs to `Unpartitioned` (the partition spec has no partition field), then the delete file is called a **global delete**. 
+Otherwise, it is called a **partition delete**.
+
+### Writing delete files
+
+From the end user's perspective, it is very rare to directly request deletion of a specific row of a specific file. 
+A delete requirement usually comes as a predicate such as `id = 5` or `date < TIMESTAMP '2000-01-01'`. 
+Given a predicate, delete files can be produced in one or some combinations of the following ways:
+
+1. **partition position deletes**: perform a scan \[1\] to know the data files and row positions affected by the predicate and then write partition position deletes \[2\]
+2. **partition equality deletes**: convert input predicate to equality predicates \[3\] for each affected partition and write partition equality deletes
+3. **partition global deletes**: convert input predicate to equality predicates and write global equality deletes 
+
+\[1\] scan here can mean a table scan, or a scan of unflushed files (stored in memory, local RocksDB, etc.) for use cases like streaming.
+
+\[2\] it is in theory possible to write global position deletes, but it is always preferred to write partition position deletes because the writer already knows the exact partition to use after the scan.
+
+\[3\] if the input inequality predicate cannot be easily converted to a finite number of equality predicates (e.g. `price > 2.33`), then it is preferred to use position deletes instead.
+
+For example, in the change data capture use case, both partition position deletes and partition equality deletes are used. 
+For a new delete such as `id = 1001`, the writer checks the unflushed data index to see if there is an existing row matching `id = 1001` and performs a position delete.
+It also writes an equality delete that is applied to all the existing data files that are already in the table.
+
+### Reading data with delete files
+
+During Iceberg's scan file planning phase, a delete file index is constructed to filter the delete files associated with any data file, with the following rules:
+
+1. equality delete files are applied to data files of strictly lower sequence numbers
```

Comment:
I still think "sequence number" needs to be defined before this section for this to make sense.

Reply:
I added a sentence at the beginning of the row-level deletes section.

Reply:
Is id the primary key? (1, 'C10', 'data10') records should only generate equality-deletes. The condition for 
pos-delete generation is that there are the same key in this commit, that is, insertedrowmap needs corresponding records to generate POS delete。

`/**
     * Write the pos-delete if there's an existing row matching the given key.
     *
     * @param key has the same columns with the equality fields.
     */
    private void internalPosDelete(StructLike key) {
      PathOffset previous = insertedRowMap.remove(key);

      if (previous != null) {
        // TODO attach the previous row if has a positional-delete row schema in appender factory.
        posDeleteWriter.delete(previous.path, previous.rowOffset, null);
      }
    }`